From 192a0e6d2df569a18bb8181b7a7fd5731766eca6 Mon Sep 17 00:00:00 2001
From: Brahma Reddy Battula <brahma@apache.org>
Date: Sat, 15 Jul 2017 10:38:31 +0800
Subject: [PATCH 2545/2748] HDFS-12112.
 TestBlockManager#testBlockManagerMachinesArray
 sometimes fails with NPE. Contributed by Wei-Chiu
 Chuang.

(cherry picked from commit b778887af59d96f1fac30cae14be1cabbdb74c8b)
(cherry picked from commit 6039a2a55f9ff0a415102f02f880c3808199a68d)

Change-Id: Iac1c0eca116dba0a87b0903362e2295d3bca9bb5
---
 .../server/blockmanagement/TestBlockManager.java   |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockManager.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockManager.java
index 5715371..82f77ca 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockManager.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockManager.java
@@ -1148,7 +1148,7 @@ public void run() {
     }
   }
 
-  @Test
+  @Test(timeout = 60000)
   public void testBlockManagerMachinesArray() throws Exception {
     final Configuration conf = new HdfsConfiguration();
     final MiniDFSCluster cluster =
@@ -1159,6 +1159,8 @@ public void testBlockManagerMachinesArray() throws Exception {
     final Path filePath = new Path("/tmp.txt");
     final long fileLen = 1L;
     DFSTestUtil.createFile(fs, filePath, fileLen, (short) 3, 1L);
+    DFSTestUtil.waitForReplication((DistributedFileSystem)fs,
+        filePath, (short) 3, 60000);
     ArrayList<DataNode> datanodes = cluster.getDataNodes();
     assertEquals(datanodes.size(), 4);
     FSNamesystem ns = cluster.getNamesystem();
-- 
1.7.9.5

